engine/apps/alerts/models/alert_group.py

```diff
@@ -133,6 +136,8 @@ def slack_templated_first_alert(self):
 
 
 class AlertGroup(AlertGroupSlackRenderingMixin, EscalationSnapshotMixin, models.Model):
+    log_records: models.QuerySet["AlertGroupLogRecord"]
```

Comment:
nit: `log_records` is not an instance of `QuerySet` (similar to my [previous comment](https://github.com/grafana/oncall/pull/2289#discussion_r1235028675))

Reply:
@vadimkerr you're right. thanks for pointing this out again and keeping me honest 😄 

so I poked around a bit and it _looks like_ this is the best approach:

```python3
if typing.TYPE_CHECKING:
    from apps.schedules.models import OnCallSchedule
    from apps.user_management.models import User

    from django.db.models.manager import RelatedManager


class Organization(MaintainableObject):
    oncall_schedules: "RelatedManager['OnCallSchedule']"

    def some_method(self):
        schedule = self.oncall_schedules.first()
        if schedule:
            schedule. # this properly resolves the typing in my IDE, see screenshot below
```
<img width="1130" alt="Screenshot 2023-06-27 at 12 08 24" src="https://github.com/grafana/oncall/assets/9406895/d759c9c3-84b1-45b1-81cf-a25db8b8d61d">

I'll go ahead and update any spots where I did this 👍

Reply:
addressed in [this commit](https://github.com/grafana/oncall/pull/2170/commits/5dc37e243acafa35c2985a58be7593f9b4b33272)